cpp/src/arrow/ipc/memory.cc

```diff
@@ -144,5 +144,29 @@ Status MemoryMappedSource::Write(int64_t position, const uint8_t* data, int64_t
   return Status::OK();
 }
 
+MockMemorySource::MockMemorySource(int64_t size) : size_(size) {}
```

Comment:
To avoid code duplication, I changed the implementation using suggestion (1) of @emkornfield; I think this means that MockMemorySource needs to be exported in the object code, please let me know if there is a way around this!